### ISSUE TITLE
Support expanding group on status page to show sub-monitors / children

### ISSUE THREAD

**Issue by stephen304** [louislam/uptime-kuma#3324]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

Now that groups are a thing, it would be nice for there to be an option on a group when editing the status page to allow expanding the group to show the details of the children. I want to use groups to make the status page less cluttered but still allow viewing the details of each child monitor without requiring admin access.

### ✔️ Solution

Add an option to the settings for a group monitor when editing a status page, enabling the setting would display a chevron icon to expand the group similar to how it expands in the admin page.

* It may make sense to disallow deleting the sub-monitors, or perhaps just allow hiding certain items if the user wishes
* It would be nice to be able to re-arrange the sub-monitors

**Comment by verymilan** [comment]
yes, please, also a grouping of some sort of the status history – like a gap between the dots (eg `|||| |||| ||||`) could be helpful for indicating at a glance that not the entire group is affected. could be a challenge tho, one of my groups got at least 13 components and likely is going to grow. I like how Vigil does this, but it would break Kumas design language.

**Comment by kawaiipantsu** [comment]
Second this feature 💯 

When clicking any Group view at least make it possible to toggle a setting to show all graphs for all sub children in the main pane below each other etc. The functionality is not only neat it's also very useful as you can "correlate" problems in a quicker way by looking at multiple graphs.

**Comment by tomerh2001** [comment]
any updates?

**Comment by CommanderStorm** [comment]
Hi there! Thanks for checking in.
The core maintainers haven't made progress on this specific issue yet, as we are currently working on a v2.0 release with many improvements, including better performance. See https://github.com/louislam/uptime-kuma/issues/4500 for more details.

If you/somebody else want to help out in this area, our [contribution guide can be found here](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md).
In case you do, please do take care that the work is compatible with other work in this area like https://github.com/louislam/uptime-kuma/pull/4419.

**Comment by CommanderStorm** [comment]
> I like how Vigil does this, but it would break Kumas design language.


@verymilan
I don't quite get your remark about this being outside our design language. (what do you mean eactly by "gap between the dots")
The screenshots for said system out there are not great (=> I don't know what you are talking about)

**Comment by teslarek** [comment]
any updates?

**Comment by CommanderStorm** [comment]
There haven’t been any updates since the last message in March — if that changes, we’ll make sure to post here. Unless you see a new comment, you can assume the status is still the same.

If you’re interested in helping define the scope or contributing a first draft, we’d be happy to support!

**Comment by thepa1n** [comment]
@CommanderStorm check please my MR with feature https://github.com/louislam/uptime-kuma/pull/6871

**Comment by CommanderStorm** [comment]
I am currently quite buisy with my real work and would ask the people in this thread to test it out if this looks like it fits into our general design and such.
